tasks/internal/request-and-upload-signature/request-and-upload-signature.yaml

```diff
@@ -118,49 +108,167 @@ spec:
           value: $(params.umb_url)
         - name: signature_data_file
           value: "$(params.signature_data_file)"
+        - name: CA_BUNDLE
+          value: /etc/pki/tls/certs/ca-bundle.crt
       script: |
         #!/usr/bin/env /bin/bash
         set -x
-        
-        MAX_RETRIES=3
-        RETRY_DELAY=5 # Initial delay
-        
-        set +x
-        # This helps with Shellcheck warning
-        echo "${UmbCert:?}" > /tmp/crt
-        echo "${UmbKey:?}" > /tmp/key
-        set -x
-        
-        echo "Requesting signing from RADAS"
-        for ((i=1; i<=MAX_RETRIES; i++)); do
-          if request-signature \
-            --manifest-digest "${manifest_digest}" \
-            --output "${signature_data_file:?}" \
-            --reference "${reference}" \
-            --requester "${requester}" \
-            --sig-key-id "${sig_key_id}" \
-            --sig-key-name "${sig_key_name}" \
-            --umb-client-name "${umb_client_name}" \
-            --umb-listen-topic "${umb_listen_topic}" \
-            --umb-publish-topic "${umb_publish_topic}" \
-            --umb-url "${umb_url}" \
-            --verbose
-          then
-            echo "request-signature command succeeded."
-            break
-          elif [ $i -eq $MAX_RETRIES ]; then
-            echo "Max retries reached. Exiting."
-            exit 1
+        cat <<EOF > "$(workspaces.data.path)/pubtools-sign-config.yaml"
```

Comment:
I wonder if it would make sense to use an emptyDir instead of the workspace - you only need to share data among steps within one task, right? It's probably not a big deal, but would be nice. If you prefer to do it afterwards, we can create a separate jira for it. But it should be fairly easy. See for example here: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/internal/push-artifacts-to-cdn-task/push-artifacts-to-cdn-task.yaml#L106

Reply:
I use the workspace for preparing values in test pipelines

Reply:
Ok, I understand how you mean it. I just updated the script

Reply:
I actually meant to get rid of the data workspace - you have both now, right? It's a bit tricky with requiring the shared workspace for the test. We said in other places that we don't want to have workspaces just for the sake of tests, but here we already had it and it would be really difficult to test this without the workspace.

So now, since you had to keep the workspace anyway, I'm not sure if it makes sense to make this change and move everything to use the emptyDir mounted to /shared. But I guess there's no harm in that. It just seems a little weird now that looking at the task definition alone, the workspace declaration seems redundant. I wonder what you think of this, would it make sense to return to the previous version? Or would you prefer to keep what you have now?